### ISSUE TITLE
Comment out blocks

### ISSUE THREAD

**Issue by pufferfish101007** [ScratchAddons/ScratchAddons#1271]
**Describe the new addon**

An addon that allows you to comment out blocks, equivalent to commenting out a line in js or python. The way I imagine it would work is:
1. Right click on block
2. Select "comment out block" in context menu
3. The block is deleted so it won't run
4. A comment is added to the parent block, or the block after if it's a hat
5. The comments text is set to something like `// sa commented-block <above/below/field xyz> <block info>
6. The comment is set not to display using css
7. A fake block is made that doesn't actually exist and is faded and added in to where the actual block should be
8. When the fake block is uncommented the fake block is deleted, the real block is added back in and the comment is deleted

It would probably be nice to have a keyboard shortcut e.g. alt click a block to (un)comment it

**Link to the new addon**

Non existent I might try making it

**Comment by RedGuy12** [comment]
But non sa would see a weird comment. Also all the blocks information would also have to be stored or else the program wouldn't know what block is supposed to go there.
________________________________

**Comment by pufferfish101007** [comment]
> But non sa would see a weird comment.

But is that a problem? It doesn't affect their experience. And in the addon info we can remind users that this is only for development and commented out blovks should be deleted gor production - like with the sa-pause custom block,it's only for development.

 > Also all the blocks information would also have to be stored or else the program wouldn't know what block is supposed to go there.

That was in the original post but gh thought it was an html tag. I've fixed the formstting now.

**Comment by CuriousG2** [comment]
I think it is a good idea, only if it can look normal for non SA users.

**Comment by Cool-Scratcher** [comment]
For non-SA users, we could replace the commented-out block with a regular comment in the workspace in place of the block that describes the block. The block would be deleted. This is just my idea, idk how practical or realistic it is. I support the idea, though.

**Comment by pufferfish101007** [comment]
> For non-SA users, we could replace the commented-out block with a regular comment in the workspace in place of the block that describes the block. The block would be deleted. This is just my idea, idk how practical or realistic it is. I support the idea, though.

That is the idea of how it's meant to work :)

**Comment by fractioneater** [comment]
I know blockly has a function for disabling blocks, so... wait... does scratch use blockly?

**Comment by CST1229** [comment]
> I know blockly has a function for disabling blocks, so... wait... does scratch use blockly?

Yes, but I don't know if they are saved with the project.json and if the VM will acknowledge the disabling.

**Comment by BroJac5246** [comment]
> > I know blockly has a function for disabling blocks, so... wait... does scratch use blockly?
> 
> Yes, but I don't know if they are saved with the project.json and if the VM will acknowledge the disabling.

It might be worth looking into, though.
